src/main/java/org/zanata/mt/model/Document.java

```diff
@@ -0,0 +1,97 @@
+package org.zanata.mt.model;
+
+import org.hibernate.annotations.NaturalId;
+import org.hibernate.validator.constraints.URL;
+
+import javax.persistence.Access;
+import javax.persistence.AccessType;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.NotNull;
+
+/**
+ * @author Alex Eng<a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Entity
+@Access(AccessType.FIELD)
+public class Document extends ModelEntity {
```

Comment:
We could have taken the opportunity to try out something other than hibernate :smiling_imp: